switchdev_utils/config-service/switchdev_network_config

```diff
@@ -0,0 +1,548 @@
+        for port in ovs_ports:
+            if port in ifcfg_interfaces_map:
+                ifcfg_interfaces_map[port] = ifcfg_interfaces_map[port] +\
+                                             port_data
+            else:
+                ovs_port_data = "%s%sDEVICE=%s" % (COMMON_IFCFG_DATA,
+                                                   port_data, port)
+                ifcfg_interfaces_map[port] = ovs_port_data
+
+
+def bind_vfs(sriov_bind_pci_list=None):
+    logger.info("Binding sriov vfs..")
+    if not sriov_bind_pci_list:
+        sriov_bind_pci_list = get_sriov_bind_pci_list()
+    for vf_pci in sriov_bind_pci_list:
+        vf_pci_driver_path = VF_PCI_PATH % {"vf_pci": vf_pci}
+        if not os.path.exists(vf_pci_driver_path):
+            try:
+                write_config(MLNX_BIND_FILE_PATH, "%s" % vf_pci)
+                logger.debug("Vf %s has been bound" % vf_pci)
+            except IOError as e:
+                logger.error("Failed to bind vf %s\n%s" % (vf_pci, str(e)))
+
+
+def configure_sriov_bind_service():
+    logger.info("Configuring sriov_bind service..")
+    write_config(SRIOV_BIND_SERVICE_FILE,
+                 SRIOV_BIND_SERVICE_CONTENT)
+    run_command('systemctl', 'enable', 'sriov_bind')
+
+
+def enable_hw_offloaded(hw_offload_enabled=False):
+    logger.info("Enabling hardware offload for openvswitch..")
+    try:
+        run_command('/usr/bin/ovs-vsctl', 'set', 'Open_Vswitch', '.',
+                    'other_config:hw-offload=%s' % hw_offload_enabled)
+    except Exception as e:
+        logger.error("Failed to set hardware offload to %s\n%s" %
+                     (hw_offload_enabled, str(e)))
+        raise
+    try:
+        run_command('systemctl', 'restart', 'openvswitch')
+    except Exception as e:
+        logger.error("Failed to restart openvswitch\n%s" % str(e))
+        raise e
+
+
+def write_ifcfg_files(ifcfg_interfaces_map):
+    logger.info("Writing ifcfg files for all net devices")
+    for interface, ifcfg_data in ifcfg_interfaces_map.items():
+        write_config(ifcfg_config_path(interface), ifcfg_data)
+
+
+def if_down_interface(device):
+    logger.info("Running /sbin/ifdown %s" % device)
+    try:
+        run_command('/sbin/ifdown', device)
+    except Exception as e:
+        logger.error("Failed to ifdown %s" % (device, str(e)))
+        raise e
+
+
+def if_up_interface(device):
+    logger.info("Running /sbin/ifup %s" % device)
+    try:
+        run_command('/sbin/ifup', device)
+    except Exception as e:
+        logger.error("Failed to ifup  %s" % (device, str(e)))
+        raise e
+
+
+def bring_interfaces_up(net_interfaces_list):
+    logger.info("Bringing interfaces down")
+    for interface in net_interfaces_list:
+        if_down_interface(interface)
+    logger.info("Bringing interfaces up")
+    for interface in net_interfaces_list:
+        if_up_interface(interface)
+
+
+def validate_config(sriov_pfs, bonds, bridges):
+    sriov_pfs_list = []
```

Comment:
i realize this method is not the heart and soul of this PR, but we should fortify it a bit more.
it will not be 100% bullet proof but will be "good enough".

if we end up exenting the config we should consider defining a schema for it